index.bs

```diff
@@ -297,6 +297,20 @@ A {{FileSystemHandle}} object is associated with a <dfn for=FileSystemHandle exp
 Note: Multiple {{FileSystemHandle}} objects can have
 [=the same locator as|the same=] [=/file system locator=].
 
+A {{FileSystemHandle}}
+<dfn for=FileSystemHandle export>is in an origin private file system</dfn>
+if the first [=list/item=] of its [=FileSystemHandle/locator=]'s
+[=file system locator/path=] is the empty string.
```

Comment:
Just kidding, I just realized that I've been confusing the incubating [Storage Buckets API](https://wicg.github.io/storage-buckets/) with the already-specified [storage bucket concept](https://storage.spec.whatwg.org/#storage-bucket).

We could give a "file system root" the option to be a "storage bottle" (?), but there's currently no way to get a bottle from the current environment AFAICT? Only to [obtain a storage bottle map](https://storage.spec.whatwg.org/#obtain-a-local-storage-bottle-map)? 

This doesn't seem to have been resolved: https://wicg.github.io/storage-buckets/#issue-0d9f1d43

Reply:
OPFS is a storage endpoint and there's one storage bottle inside a storage bucket for each storage endpoint. (The endpoint is like a class, whereas the bottle is an instance of that class.) A storage shelf equates to a storage partition defined by a StorageKey. So a shelf contains buckets which contains bottles.

The issue mentioned in the storage buckets spec is that we need a version of [this algorithm](https://whatpr.org/fs/110.html#dom-storagemanager-getdirectory) which works for any bucket. You'll notice if you drill down into "obtain a local storage bottle map" that it currently assumes the "default" bucket. This just means passing an additional parameter.

We do not currently have a way to go from bottle to bucket for any storage endpoint.

Reply:
Thanks for the clarification @estade. I filed #112 to track making `getDirectory()` work for non-default buckets. Addressing https://github.com/whatwg/storage/issues/2 seems like a prerequisite for adding another parameter to "obtain a local storage bottle map"?

> We do not currently have a way to go from bottle to bucket for any storage endpoint.

We just need some way to tie a FileSystemHandle to its root (i.e. two files of the same "file system path" must be considered different if they're from different buckets). There's one OPFS instance per storage bucket, though since there's also only one "fileSystem" bottle per bucket, then using the bottle would also work... though currently, it seems like it's only possible to get a "storage bottle map" and not the storage bottle itself. It seems a bit clumsy to associate a "storage bottle map" with each "file system locator"

Which raises the question of... why don't we just point to the directory entry at `map["root"]` directly? That would essentially make every "file system locator" a relative path, which on one hand could improve the hand-waviness of the recently-added "[locate an entry](https://fs.spec.whatwg.org/#locating-an-entry)" and "[get the locator](https://fs.spec.whatwg.org/#getting-the-locator)" methods, but on the other hand re-raises some of the questions of #59 (i.e. what happens if a root is deleted or moved? https://github.com/whatwg/fs/pull/9#discussion_r1045021650)

So, for now at least, I'd prefer to keep "file system locators" absolute and find a way to point to a bottle/bucket :)

Reply:
I don't think there's any reason not to have a way to retrieve a bottle. We do have ways to get a bucket, but they're currently specified [here](https://pr-preview.s3.amazonaws.com/evanstade/storage-buckets/pull/88.html#storage-bucket-open) instead of in the storage spec.

Reply:
I think if we define OPFS in terms of https://storage.spec.whatwg.org/#obtain-a-local-storage-bottle-map you could make this rely less on magic. But that's a more substantial overhaul.

Reply:
Ack. I'll leave this PR as just specifying OPFS-ness on a `FileSystemHandle`, then we can leave better defining the OPFS itself (in terms of the Storage spec) to a follow-up